### ISSUE TITLE
error trying to install scipy in HASS.io

### ISSUE THREAD

**Issue by WillyWachendon** [AppDaemon/appdaemon#1632]
I need to use the module scipy in my project, I have added scipy to the requirements. Looking at the logs I see this 

Requirement already satisfied: six>=1.5 in /usr/lib/python3.10/site-packages (from python-dateutil>=2.8.1->pandas) (1.16.0)
Installing collected packages: numpy, pandas
Successfully installed numpy-1.24.1 pandas-1.4.4
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
Looking in links: https://wheels.home-assistant.io/musllinux/
Collecting scipy
  Downloading scipy-1.10.0.tar.gz (42.4 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.4/42.4 MB 11.5 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Installing backend dependencies: started
  Installing backend dependencies: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [55 lines of output]
      + meson setup --prefix=/usr /tmp/pip-install-8i6pmzcv/scipy_2ab0e0e18b90413697eac73cdb22516e /tmp/pip-install-8i6pmzcv/scipy_2ab0e0e18b90413697eac73cdb22516e/.mesonpy-0qzvbhwy/build --native-file=/tmp/pip-install-8i6pmzcv/scipy_2ab0e0e18b90413697eac73cdb22516e/.mesonpy-native-file.ini -Ddebug=false -Doptimization=2
      The Meson build system
      Version: 1.0.0
      Source dir: /tmp/pip-install-8i6pmzcv/scipy_2ab0e0e18b90413697eac73cdb22516e
      Build dir: /tmp/pip-install-8i6pmzcv/scipy_2ab0e0e18b90413697eac73cdb22516e/.mesonpy-0qzvbhwy/build
      Build type: native build
      Project name: SciPy
      Project version: 1.10.0
      
      ../../meson.build:1:0: ERROR: Unknown compiler(s): [['cc'], ['gcc'], ['clang'], ['nvc'], ['pgcc'], ['icc'], ['icx']]
      The following exception(s) were encountered:
      Running `cc --version` gave "[Errno 2] No such file or directory: 'cc'"
      Running `gcc --version` gave "[Errno 2] No such file or directory: 'gcc'"
      Running `clang --version` gave "[Errno 2] No such file or directory: 'clang'"
      Running `nvc --version` gave "[Errno 2] No such file or directory: 'nvc'"
      Running `pgcc --version` gave "[Errno 2] No such file or directory: 'pgcc'"
      Running `icc --version` gave "[Errno 2] No such file or directory: 'icc'"
      Running `icx --version` gave "[Errno 2] No such file or directory: 'icx'"
      
      A full log can be found at /tmp/pip-install-8i6pmzcv/scipy_2ab0e0e18b90413697eac73cdb22516e/.mesonpy-0qzvbhwy/build/meson-logs/meson-log.txt
      Traceback (most recent call last):
        File "/usr/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 144, in prepare_metadata_for_build_wheel
          hook = backend.prepare_metadata_for_build_wheel
      AttributeError: module 'mesonpy' has no attribute 'prepare_metadata_for_build_wheel'
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "/usr/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
          main()
        File "/usr/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/usr/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 148, in prepare_metadata_for_build_wheel
          whl_basename = backend.build_wheel(metadata_directory, config_settings)
        File "/tmp/pip-build-env-wx5z0cdp/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 1060, in build_wheel
          with _project(config_settings) as project:
        File "/usr/lib/python3.10/contextlib.py", line 135, in __enter__
          return next(self.gen)
        File "/tmp/pip-build-env-wx5z0cdp/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 975, in _project
          with Project.with_temp_working_dir(
        File "/usr/lib/python3.10/contextlib.py", line 135, in __enter__
          return next(self.gen)
        File "/tmp/pip-build-env-wx5z0cdp/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 750, in with_temp_working_dir
          yield cls(source_dir, tmpdir, build_dir, meson_args)
        File "/tmp/pip-build-env-wx5z0cdp/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 632, in __init__
          self._configure(reconfigure=bool(build_dir) and not native_file_mismatch)
        File "/tmp/pip-build-env-wx5z0cdp/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 680, in _configure
          self._meson('setup', *setup_args)
        File "/tmp/pip-build-env-wx5z0cdp/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 657, in _meson
          return self._proc('meson', *args)
        File "/tmp/pip-build-env-wx5z0cdp/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 652, in _proc
          subprocess.check_call(list(args), env=self._env)
        File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['meson', 'setup', '--prefix=/usr', '/tmp/pip-install-8i6pmzcv/scipy_2ab0e0e18b90413697eac73cdb22516e', '/tmp/pip-install-8i6pmzcv/scipy_2ab0e0e18b90413697eac73cdb22516e/.mesonpy-0qzvbhwy/build', '--native-file=/tmp/pip-install-8i6pmzcv/scipy_2ab0e0e18b90413697eac73cdb22516e/.mesonpy-native-file.ini', '-Ddebug=false', '-Doptimization=2']' returned non-zero exit status 1.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
× Encountered error while generating package metadata.
╰─> See above for output.
note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

Could someone help me with the problem?

**Comment by eli-schwartz** [comment]
> I need to use the module scipy in my project, I have added scipy to the requirements. Looking at the logs I see this

The error is, as you noted, an attempt to build SciPy -- nothing to do with appdaemon. For help building SciPy, you should see https://scipy.org/install/ -- note it explicitly warns you that building binaries from source is "nontrivial". There are links there to the developer documentation for accomplishing this anyway.

tl;dr -- your current error is:

```
  ../../meson.build:1:0: ERROR: Unknown compiler(s): [['cc'], ['gcc'], ['clang'], ['nvc'], ['pgcc'], ['icc'], ['icx']]
  The following exception(s) were encountered:
  Running `cc --version` gave "[Errno 2] No such file or directory: 'cc'"
  Running `gcc --version` gave "[Errno 2] No such file or directory: 'gcc'"
  Running `clang --version` gave "[Errno 2] No such file or directory: 'clang'"
  Running `nvc --version` gave "[Errno 2] No such file or directory: 'nvc'"
  Running `pgcc --version` gave "[Errno 2] No such file or directory: 'pgcc'"
  Running `icc --version` gave "[Errno 2] No such file or directory: 'icc'"
  Running `icx --version` gave "[Errno 2] No such file or directory: 'icx'"
```

Which is saying you need a Fortran compiler installed, but don't have one.

This is described at https://scipy.github.io/devdocs/dev/contributor/building.html#system-level-dependencies

**Comment by WillyWachendon** [comment]
Thanks a lot @eli-schwartz. It helps me to understand. Do you see any solution to fix the installation in Hassio? As AppDaemon Addon I dont have much control on what I can or how to install components, the only options are the Add-on configurations, and as far as I see them, are limited to the name of the module.

**Comment by WillyWachendon** [comment]
Finally I found a solution, I got access to the homeassistant container and installed the alpine equivalent py3-scipy package, then I added symbolic links to usr/local/bin/python3.10/scipy folder
